eclipse-theia/theia
# 8969-upgrade-monaco-editor-core-to-standalone-0.23.x

In preparation for upgrading monaco-editor-core to standalone/0.23.x these changes were implemented:

1. signatures alignment to standalone/0.23.x
2. editor preferences synced with standalone/0.23.x

CQ: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=23341

Signed-off-by: Dan Arad dan.arad@sap.com

#### What it does
This PR updates Theia repo with these changes:

1) signatures alignment to standalone/0.23.x
2) editor preferences synced with standalone/0.23.x
3) Changes to theme services based on changes in 0.23.x 
4) Removal of QuickOpen modules and implementing QuickInput instead.
5) Running quickInput functionality in Extensions - this I am still currently being worked on by me


(It is also second part of 8969 where first part is generating monaco-editor-core@0.23)

#### How to test
1. How to test section for the previous PR #8010.
2. Integration tests in: #7265.
3. All quickOpen replaced by quickInput functionality like:

- All Quick Open prefixes (Ctrl+P and then '?') - also from shortcut keys:
 ![image](https://user-images.githubusercontent.com/71069170/116989935-18dec500-acdb-11eb-8732-18cc5d185300.png)
- All GIT commands - Go to source control view (Ctrl+Shift+G) and test out all git commands including setting up new repository   cloning , fetch, pull, push, merge, checkout, chooseTagsAndBranches , commitMessageForAmend, stash actions, also **GIT ASKPASS** in electron (which I’m not sure yet how to test) and also git sync functionality (theia\packages\git\src\browser\git-sync-service.ts). Also Change repository (ScmQuickOpenService):.
 -- there is also electron-browser/prompt/git-quick-open-prompt.ts which needs testing - and I'm not sure how to test it.
- mini browser - open url
-  all tasks functionality

4) All functionality of themes should be tested (Select File Icon Theme, Select Color Theme)
5) Theia Commands:

- Choosing keyboard layout: “core.keyboard.choose”
- “Editor: Change Language Mode”
- “Editor: Change File Encoding”
- Config indentation: 'textEditor.commands.configIndentation'
- 'Indent Using Spaces'
- 'Indent Using Tabs'
- Selecting formatter for document (when there are multiple ones) – i.e.: 
![image](https://user-images.githubusercontent.com/71069170/117776607-7081b100-b244-11eb-88f2-55376d46c5ea.png)
 
- 'Deploy Plugin by Id'
- Using “vscode-extension-samples/configuration-sample” sample, run 'config.commands.configureEmptyLastLineFiles' command and select workspace folder:
![image](https://user-images.githubusercontent.com/71069170/117776583-695aa300-b244-11eb-9b32-3c0df7475571.png)

- Register Variables – for VariableContribution.
- ‘Open Recent Workspace’

6.	Quick Input extensions usage – using “vscode-extension-samples/quickinput-sample” sample.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
